compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -1258,6 +1263,16 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
                         candidates.remove(idx);
                     }
 
+                    let is_stable =
+                        is_stable && did.is_some() && this.is_stable(did.unwrap(), path.span);
+                    if is_stable
+                        && let Some(idx) = candidates
+                            .iter()
+                            .position(|v: &ImportSuggestion| v.did == did && !v.is_stable)
```

Comment:
I don't understand the logic here, what does this condition check - an unstable reexport of a stable item? Could you add a comment?
Also, there may be several such reexports, they should probably all be removed.
Also, the PR description doesn't really match this logic.

Reply:
> 1. yes, it removes unstable ones if exist. e.g., `std::range::legacy::Range` vs `std::ops::Range`, `legacy` in the former is unstable, we prefer to suggest `use` the latter. And only do this check if current one is stable
> 2. there could only be one cause same items will only be inserted once, see [line 1284](https://github.com/rust-lang/rust/pull/140795/files#diff-0e459b4cbb41dd93274d7c68b18a2a3eab7152e9aea107839a7bddff0b4e9195R1284)
> 3. so the logic is to replace unstable suggestions if we meet a new stable one, and do nothing if any other situation, I think it matches the PR description. But I'm not a native English speaker, so maybe you have a better suggestion? ;)

An explanation like this is exactly what I'd expect to seen in the PR desctiption and code comments.